-1

Tested at: 8b93d100c7f2fd0c860bb98458ca1c2ca39b4255

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ecd997/2164/summary.html

I found follow errors while testing this PR

**Failed tests:  ClangBuild**

* **Clang**:

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```


<pre>>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-08-23-2300/src/RecoLocalMuon/DTSegment/src/DTSegment4DT0Corrector.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-08-23-2300/src/RecoLocalMuon/DTSegment/src/DTRefitAndCombineReco4D.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-08-23-2300/src/RecoLocalMuon/DTSegment/src/DTRecSegment2DAlgoFactory.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-08-23-2300/src/RecoLocalMuon/DTSegment/src/DTClusterer.cc 
>> Compiling  /build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-08-23-2300/src/RecoLocalMuon/DTSegment/src/DTCombinatorialPatternReco.cc 
/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-08-23-2300/src/RecoLocalMuon/DTSegment/src/DTSegmentUpdator.cc:542:19: error: variable-sized object may not be initialized
  float residuals[N] = {};
                  ^
1 error generated.
gmake: *** [tmp/slc7_amd64_gcc700/src/RecoLocalMuon/DTSegment/src/RecoLocalMuonDTSegment/DTSegmentUpdator.cc.o] Error 1
Entering library rule at src/RecoLocalMuon/DTSegment/test
</pre>

--- a/RecoLocalMuon/DTSegment/src/DTSegmentUpdator.cc
+++ b/RecoLocalMuon/DTSegment/src/DTSegmentUpdator.cc
@@ -495,7 +495,7 @@ void DTSegmentUpdator::rejectBadHits(DTChamberRecSegment2D* phiSeg) const {
     cout << " Inside the segment updator, now loop on hits:   ( x == z_loc , y == x_loc) " << endl;
 
   vector<DTRecHit1D> hits = phiSeg->specificRecHits();
-  size_t N = hits.size();
+  const size_t N = hits.size();
   if (N < 3)
     return;
 
@@ -539,7 +539,7 @@ void DTSegmentUpdator::rejectBadHits(DTChamberRecSegment2D* phiSeg) const {
     cout << "fit 2 parameters done ----> par0: " << par[0] << "  par1: " << par[1] << endl;
 
   // Calc residuals:
-  float residuals[N] = {};
+  float residuals[N];
   float mean_residual = 0.;  //mean of the absolute values of residuals
   for (size_t i = 0; i < N; ++i) {
     residuals[i] = y.at(i) - par[1] * x.at(i) - par[0];

